File: typegate/tests/runtimes/deno/deno_duplicate_artifact.py
import os
import sys

from typegraph.gen.exports.core import (
    ArtifactResolutionConfig,
    MigrationAction,
    MigrationConfig,
)
from typegraph.graph.shared_types import BasicAuth
from typegraph.graph.tg_deploy import TypegraphDeployParams, tg_deploy
from typegraph.graph.typegraph import Graph
from typegraph.policy import Policy
from typegraph.runtimes.deno import DenoRuntime

from typegraph import t, typegraph


@typegraph()
def deno_duplicate_artifact(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    g.expose(
        public,
        doAddition=deno.import_(
            t.struct({"a": t.float(), "b": t.float()}),
            t.float(),
            module="ts/dep/main.ts",
            deps=["ts/dep/nested/dep.ts"],
            name="doAddition",
        ),
        doAdditionDuplicate=deno.import_(
            t.struct({"a": t.float(), "b": t.float()}),
            t.float(),
            module="ts/dep/main.ts",
            deps=["ts/dep/nested/dep.ts"],
            name="doAdditionDuplicate",
        ),
    )


cwd = sys.argv[1]
PORT = sys.argv[2]
gate = f"http://localhost:{PORT}"
auth = BasicAuth("admin", "password")

deno_tg = deno_duplicate_artifact()
deploy_result = tg_deploy(
    deno_tg,
    TypegraphDeployParams(
        base_url=gate,
        auth=auth,
        typegraph_path=os.path.join(cwd, "deno_duplicate_artifact.py"),
        artifacts_config=ArtifactResolutionConfig(
            dir=cwd,
            prefix=None,
            disable_artifact_resolution=None,
            codegen=None,
            prisma_migration=MigrationConfig(
                migration_dir="prisma-migrations",
                global_action=MigrationAction(reset=False, create=True),
                runtime_actions=None,
            ),
        ),
    ),
)

print(deploy_result.serialized)

File: typegate/tests/runtimes/deno/deno_duplicate_artifact.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.js";

export const denoDuplicateArtifact = await typegraph({
  name: "test_deno_dep",
}, (g) => {
  const deno = new DenoRuntime();
  const pub = Policy.public();

  g.expose({
    doAddition: deno.import(
      t.struct({ a: t.float(), b: t.float() }),
      t.float(),
      {
        module: "ts/dep/main.ts",
        name: "doAddition",
        deps: ["ts/dep/nested/dep.ts"],
      },
    ).withPolicy(pub),
    doAdditionDuplicate: deno.import(
      t.struct({ a: t.float(), b: t.float() }),
      t.float(),
      {
        module: "ts/dep/main.ts",
        name: "doAddition",
        deps: ["ts/dep/nested/dep.ts"],
      },
    ).withPolicy(pub),
  });
});
